File: 09-20-introduction/fetch-decode-execute-cycle/c/main.c
#include <time.h>
#include <stdio.h>

#define ITERS 1000000000

int main (int argc, char** argv) {
    clock_t start = clock();
    for (int i = 0; i < ITERS; i++);
    float secs = (float)(clock() - start) / CLOCKS_PER_SEC;
    float ops = ITERS / secs;
    printf("Elapsed: %.3f\n", secs);
    printf("Clock speed approx %.3f GHz\n", ops / 1000000000.0);
}

File: 09-20-introduction/fetch-decode-execute-cycle/rust/src/main.rs
use std::time::Instant;

const ITERS: u128 = 1_000_000_000;

fn main() {
    let now = Instant::now();
    let mut i = 0;
    loop {
        i += 1;
        if i > ITERS {
            break;
        }
    }
    let elapsed = now.elapsed();
    println!("elapsed: {:?}", elapsed);
    let secs = elapsed.as_secs_f64();
    let ops = ITERS as f64 / secs;
    println!("clock speed approx {} GHz", ops / 1_000_000_000.0);
}
